pkg/volume/csi/csi_plugin.go

```diff
@@ -752,6 +760,21 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) {
 		return false, nil
 	}
+
+	kHost, ok := p.host.(volume.KubeletVolumeHost)
```

Comment:
skipAttach is called by both kubelet and AD controller.

Can we add the HasSynced to `csiPlugin` struct similarly to csiDriverLister?  And then add a helper method that the skipAttach and podInfoOnMount calls to both call to wait for sync?

Reply:
AD controller already has the WaitForCacheSync logic for csiDriversSynced in Run().  That's why I skipped it.

Sure, I'll make a helper method.